core-bundle/contao/forms/Form.php

```diff
@@ -560,10 +587,41 @@ protected function processFormData($arrSubmitted, $arrLabels, $arrFields, $arrFi
 			System::getContainer()->get('monolog.logger.contao.forms')->info('Form "' . $this->title . '" has been submitted by a guest.');
 		}
 
+		$targetPageData = null;
+
 		// Check whether there is a jumpTo page
 		if (($objJumpTo = $this->objModel->getRelated('jumpTo')) instanceof PageModel)
 		{
-			$this->jumpToOrReload($objJumpTo->row());
+			$targetPageData = $objJumpTo->row();
+		}
+
+		// Set the confirmation message, if any
+		if ($this->objModel->confirmationMessage)
+		{
+			$confirmationMessage = $this->objModel->confirmationMessage;
+			$confirmationMessage = System::getContainer()->get('contao.string.simple_token_parser')->parse($confirmationMessage, $arrSubmitted);
+			$confirmationMessage = System::getContainer()->get('contao.insert_tag.parser')->replaceInline($confirmationMessage);
+
+			$requestStack = System::getContainer()->get('request_stack');
+			$request = $requestStack->getCurrentRequest();
+
+			// Throw the response exception if it's an AJAX request
+			if ($request && $request->isXmlHttpRequest() && $request->headers->get('X-Contao-Ajax-Form') === $this->getFormId() && $targetPageData === null)
+			{
+				$confirmationTemplate = new FrontendTemplate('form_confirmation');
+				$confirmationTemplate->setData($this->Template->getData());
+				$confirmationTemplate->confirmation = $confirmationMessage;
```

Comment:
```suggestion
				$confirmationTemplate->confirmation = StringUtil::specialchars($confirmationMessage);
```

Reply:
The confirmation message field contains HTML (tinyMCE).